docs/compiling.rst

```diff
@@ -201,6 +194,17 @@ to an independently constructed (through ``add_library``, not
     Studio (``/bigobj``).  The :ref:`FAQ <faq:symhidden>` contains an
     explanation on why these are needed.
 
+    If you want to add these in yourself, you can use:
+
+    .. code-block:: cmake
+
+        cmake_minimum_required(3.9)
+        set(CMAKE_CXX_VISIBILITY_PRESET hidden)
```

Comment:
What about `VISIBILITY_INLINES_HIDDEN`, should we put that in here too?

Reply:
Do we need to add symbol hiding ourselves? I would somehow like if this kept being the default, as it is now. I think lots of users might have no idea about it, but would produce huge extension modules/wheels. Other, running into problems with visibility, probably are doing more intricate stuff, so should be able to do research and disable "hidden" visibility, I'd say.

Reply:
Summary from below (not that GitHub lets me comment on this again): the was only for the "advanced" target (regular does proper visibility hiding still), and there are strong downsides to trying to set visibility through an interface target (unless CMake adds first class support for it), and it damages the "good" usage of it via the `pybind11_add_module` function.

I should note  here that 3.9+ is only needed for the line below, about the LTO; CMake 3.7 handles the VISIBILITY just fine.